internal/codegen/golang/templater.go

```diff
@@ -175,7 +175,7 @@ func (tm Templater) templateFile(file codegen.QueryFile, isLeader bool) (Templat
 
 		nonVoidCols := removeVoidColumns(outputs)
 		resultKind := query.ResultKind
-		if len(nonVoidCols) == 0 {
+		if len(nonVoidCols) == 0 && resultKind != ast.ResultKindString {
```

Comment:
Bug fix: pggen has this idea that if nothing is returned you may as well treat it like `:exec` but `:string` shouldn't work this way.